packages/auth/src/Authenticator.js

```diff
@@ -17,6 +19,39 @@ export class Authenticator {
     this.accessPolicyBuilder = new AccessPolicyBuilderClass(models);
   }
 
+  /**
+   * Authenticate by access token claims
+   * @param {{ userId: string, apiClientUserId?: string }} accessTokenClaims
+   */
+  async authenticateAccessToken({ userId, apiClientUserId }) {
+    const user = await this.models.user.findById(userId);
+    const userAccessPolicy = await this.getAccessPolicyForUser(user.id);
+
+    if (!apiClientUserId) {
+      return { user, accessPolicy: userAccessPolicy };
+    }
+
+    const apiClientAccessPolicy = await this.getAccessPolicyForUser(apiClientUserId);
+    return { user, accessPolicy: mergeAccessPolicies(userAccessPolicy, apiClientAccessPolicy) };
+  }
+
+  /**
+   * Authenticate an api client user
+   * @param {{ username: string, secretKey: string }} apiClientCredentials
+   */
+  async authenticateApiClient({ username, secretKey }) {
```

Comment:
This logic here isn't quite the same as `src/auth/clientAuth.js`. The `clientAuth` version supports the `env.CLIENT_USERNAME, env.CLIENT_SECRET` user credentials (which don't actually exist in the database...).

Personally I'd love to add that api client user to the database as it would allow us to remove such branching logic, but am wary of the regression risk. If the reviewer agrees, I'll create a ticket for it and we can address when there's time 👍 

For now is it acceptable to re-introduce this logic here (basically I've pulled it out from `@tupaia/server-boilerplate`)

Reply:
Yes totally, after your change to make web-config-server use a db based api client it's just meditrak-app using the one that's in env vars. That's just a bit of laziness because we introduced db based api clients at a high pressure time and didn't bother to add in the existing api client, but please do create a ticket for that (this comment shows we meant to do it pretty quickly, but just never did)
```
  // Still check against environment variables so that these
  // updates can be deployed without breaking the app. This
  // check can be removed once the proper credentials have been
  // added to the production database.
```

Reply:
Created: https://github.com/beyondessential/tupaia-backlog/issues/2884